iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/IoTDBShutdownHook.java

```diff
@@ -49,8 +50,11 @@ public class IoTDBShutdownHook extends Thread {
 
   private static final Logger logger = LoggerFactory.getLogger(IoTDBShutdownHook.class);
 
-  public IoTDBShutdownHook() {
+  private final TDataNodeLocation nodeLocation;
+
+  public IoTDBShutdownHook(TDataNodeLocation nodeLocation) {
```

Comment:
How about rename this class to DataNodeShutDownHook? @OneSizeFitsQuorum what do you think

Reply:
This PR is irrelevant  with here. We can do it in next PR.

Reply:
lgtm